catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -159,23 +168,24 @@ public Metacard getMetacard(String id)
         return getMetacards(Metacard.ID, ids, tagFilter);
     }
 
-    public Map<String, Result> getMetacards(Collection<String> ids, Filter tagFilter)
+    public Map<String, Result> getMetacards(List<String> writableSources, Collection<String> ids, Filter tagFilter)
             throws UnsupportedQueryException, SourceUnavailableException, FederationException {
-        return getMetacards(Metacard.ID, ids, tagFilter);
+        return getMetacards(writableSources, Metacard.ID, ids, tagFilter);
     }
 
     public Map<String, Result> getMetacards(String attributeName, Collection<String> ids,
             String tag)
             throws UnsupportedQueryException, SourceUnavailableException, FederationException {
-        return getMetacards(attributeName,
+        return getMetacards(null,
+                attributeName,
                 ids,
                 filterBuilder.attribute(Metacard.TAGS)
                         .is()
                         .like()
                         .text(tag));
     }
 
-    public Map<String, Result> getMetacards(String attributeName, Collection<String> ids,
+    public Map<String, Result> getMetacards(@Nullable List<String> writableSources, String attributeName, Collection<String> ids,
```

Comment:
✏️  Sorry, I didn't realize this was a `public` method in the first round of review when I asked for the `@Nullable`. As a general rule we try to avoid `null` as a valid value in public methods to prevent unexpected NPEs. In this case, if callers don't have a list of `writableSources`, they could pass `Collections.emptyList()`, which is a singleton list that doesn't need to be instantiated.